RootCommandBuilder.cs

```diff
@@ -0,0 +1,47 @@
+using System.Net.Http;
+using System.CommandLine;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using System.CommandLine.Invocation;
+using System.Threading;
+using System;
+
+namespace project
+{
+    public class RootCommandBuilder
+    {
+        private readonly IEnumerable<IFeature> features;
+
+        public RootCommandBuilder(IEnumerable<IFeature> features)
+        {
+            this.features = features;
+        }
+
+        public RootCommand BuildRootCommand()
+        {
+            var rootCommand = new RootCommand("curl-like command line tool to transfer data to and from servers");
+
+            foreach (var feature in features)
+            {
+                feature.Activate(rootCommand);
```

Comment:
Each "feature" gets a change to add some option or some parameter to the root command